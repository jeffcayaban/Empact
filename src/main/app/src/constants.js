export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;