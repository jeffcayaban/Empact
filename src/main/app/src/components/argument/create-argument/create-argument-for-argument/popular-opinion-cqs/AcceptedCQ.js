import React from 'react'
import { Panel } from "react-bootstrap";
import CreateArgumentOptions from "../../CreateArgumentOptions";
import {ARGUMENTATION_SCHEMES, CRITICAL_QUESTION_IDS} from "../../../../../constants";
import {getIsArgumentAgreeWithCQ, showArgumentTemplate} from "../../../../../utils/Helpers";
import UserOpinionBtn from "../../../../common/UserOpinionBtn";
import ExplanationPreviewModal from "../../../../common/ExplanationPreviewModal";

/**
 * Will display critical question "Is the explanation generally accepted?" and provide options to create an argument.
 */

class AcceptedCQ extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            showOptions: false,
            showExplanation: false,
            selectedArgScheme: ''
        };

        this.explanationBtnCallbackFn = this.explanationBtnCallbackFn.bind(this);
        this.handleShowArgumentOptions = this.handleShowArgumentOptions.bind(this);
    }

    onOptionSelect = (optionKey) => this.setState({ selectedArgScheme: ARGUMENTATION_SCHEMES[optionKey]});
    handleShowArgumentOptions = () => this.setState({ showOptions: !this.state.showOptions });
    explanationBtnCallbackFn = () => this.setState({ showExplanation: !this.state.showExplanation });

    render() {
        const { isSupporting, argument } = this.props;
        const { selectedArgScheme, showOptions } = this.state;
        const isOrNot = isSupporting ? '' : 'not ';
        const isArgumentAgreeWithCQ = getIsArgumentAgreeWithCQ(isSupporting, argument.agreesWithCQ);

        const argumentTemplates = showArgumentTemplate(null, selectedArgScheme, null, isSupporting,
            isArgumentAgreeWithCQ, argument.id, argument.criticalQuestionId, CRITICAL_QUESTION_IDS[3]);

        const explanationBtn = (
            <button type="button" className="linkButton" onClick={this.explanationBtnCallbackFn}>
                explanation
            </button>
        );

        return (
            <Panel expanded={showOptions}>
                <Panel.Heading className={"argumentAssumption"}>
                    <div>
                        {`1. Can you provide a supporting argument to prove that the `}
                        { explanationBtn }
                        {` is ${isOrNot}generally accepted?`}
                    </div>
                    <UserOpinionBtn showOption={showOptions} showArgTemplates={this.handleShowArgumentOptions} />
                </Panel.Heading>
                <Panel.Collapse>
                    <Panel.Body>
                        <CreateArgumentOptions handleSelect={this.onOptionSelect.bind(this)} />
                        { argumentTemplates }
                    </Panel.Body>
                </Panel.Collapse>
                <ExplanationPreviewModal
                    showExplanation={this.state.showExplanation}
                    closeCallbackFn={this.explanationBtnCallbackFn}
                    explanation={argument.explanation}
                    container={this}
                />
            </Panel>
        );
    }
}

export default AcceptedCQ;
