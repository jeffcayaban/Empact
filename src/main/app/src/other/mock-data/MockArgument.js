import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};