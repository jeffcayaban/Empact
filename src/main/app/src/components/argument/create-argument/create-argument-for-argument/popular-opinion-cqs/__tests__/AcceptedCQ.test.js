import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import {ARGUMENTATION_SCHEMES } from "../../../../../../constants";
import AcceptedCQ from "../AcceptedCQ";

import MockArgument from '../../../../../../other/mock-data/MockArgument';

describe('AcceptedCQ', () => {

    it('should render correctly', () => {
        const component = shallow(<AcceptedCQ argument={MockArgument} isSupporting={true} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should call handleShowArgumentOptions', () => {
        const component = shallow(<AcceptedCQ argument={MockArgument} isSupporting={true} />);
        const instance = component.instance();

        const oldShowOptionsValue = component.state().showOptions;
        instance.handleShowArgumentOptions();
        const newShowOptionsValue = component.state().showOptions;

        expect(newShowOptionsValue).not.toBe(oldShowOptionsValue);
        expect(newShowOptionsValue).toBe(true)
    });

    it('should call onOptionSelect', () => {
        const component = shallow(<AcceptedCQ argument={MockArgument} isSupporting={true} />);
        const instance = component.instance();
        instance.onOptionSelect(0);
        const newSelectedArgValue = component.state().selectedArgScheme;
        expect(newSelectedArgValue).toBe(ARGUMENTATION_SCHEMES[0])
    });

    it('should call explanationBtnCallbackFn', () => {
        const component = shallow(<AcceptedCQ argument={MockArgument} isSupporting={true} />);
        const instance = component.instance();

        const oldShowExplanationValue = component.state().showExplanation;
        instance.explanationBtnCallbackFn();
        const newShowExplanationValue = component.state().showExplanation;

        expect(newShowExplanationValue).not.toBe(oldShowExplanationValue);
        expect(newShowExplanationValue).toBe(true)
    });
});
