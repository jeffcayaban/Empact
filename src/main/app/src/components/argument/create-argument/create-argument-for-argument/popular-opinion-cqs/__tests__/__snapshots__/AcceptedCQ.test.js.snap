// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptedCQ should render correctly 1`] = `
<Uncontrolled(Panel)
  expanded={false}
>
  <PanelHeading
    bsClass="panel"
    className="argumentAssumption"
    componentClass="div"
  >
    <div>
      1. Can you provide a supporting argument to prove that the 
      <button
        className="linkButton"
        onClick={[Function]}
        type="button"
      >
        explanation
      </button>
       is generally accepted?
    </div>
    <UserOpinionBtn
      showArgTemplates={[Function]}
      showOption={false}
    />
  </PanelHeading>
  <PanelCollapse
    bsClass="panel"
  >
    <PanelBody
      bsClass="panel"
      collapsible={false}
    >
      <CreateArgumentOptions
        handleSelect={[Function]}
      />
    </PanelBody>
  </PanelCollapse>
  <ExplanationPreviewModal
    closeCallbackFn={[Function]}
    container={
      AcceptedCQ {
        "context": Object {},
        "explanationBtnCallbackFn": [Function],
        "handleShowArgumentOptions": [Function],
        "onOptionSelect": [Function],
        "props": Object {
          "argument": Object {
            "agreesWithCQ": true,
            "argumentType": "ExpertOpinion",
            "createdBy": Object {
              "authorities": Array [
                "ROLE_ADMIN",
              ],
              "firstName": "testFirstName",
              "id": "testId",
              "lastName": "testLastName",
              "username": "testUsername",
            },
            "creationDateTime": 1551716401541,
            "criticalQuestionId": "CQ1",
            "expert": "testExpert",
            "expertAssertion": "testAssertion",
            "expertDomain": "testDomain",
            "id": "testId",
            "isAnonymous": false,
            "isSupporting": true,
            "lastUpdatedDateTime": 1551716402000,
            "parentArgumentId": "testParentArgumentId",
            "rootPetitionId": "testRootPetitionId",
            "sources": Array [],
            "subCriticalQuestionId": null,
          },
          "isSupporting": true,
        },
        "refs": Object {},
        "setState": [Function],
        "state": Object {
          "selectedArgScheme": "",
          "showExplanation": false,
          "showOptions": false,
        },
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_currentlyRenderingComponent": null,
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <AcceptedCQ
              argument={
                Object {
                  "agreesWithCQ": true,
                  "argumentType": "ExpertOpinion",
                  "createdBy": Object {
                    "authorities": Array [
                      "ROLE_ADMIN",
                    ],
                    "firstName": "testFirstName",
                    "id": "testId",
                    "lastName": "testLastName",
                    "username": "testUsername",
                  },
                  "creationDateTime": 1551716401541,
                  "criticalQuestionId": "CQ1",
                  "expert": "testExpert",
                  "expertAssertion": "testAssertion",
                  "expertDomain": "testDomain",
                  "id": "testId",
                  "isAnonymous": false,
                  "isSupporting": true,
                  "lastUpdatedDateTime": 1551716402000,
                  "parentArgumentId": "testParentArgumentId",
                  "rootPetitionId": "testRootPetitionId",
                  "sources": Array [],
                  "subCriticalQuestionId": null,
                }
              }
              isSupporting={true}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_previousComponentIdentity": null,
            "_renderPhaseUpdates": null,
            "_rendered": <Uncontrolled(Panel)
              expanded={false}
            >
              <PanelHeading
                bsClass="panel"
                className="argumentAssumption"
                componentClass="div"
              >
                <div>
                  1. Can you provide a supporting argument to prove that the 
                  <button
                    className="linkButton"
                    onClick={[Function]}
                    type="button"
                  >
                    explanation
                  </button>
                   is generally accepted?
                </div>
                <UserOpinionBtn
                  showArgTemplates={[Function]}
                  showOption={false}
                />
              </PanelHeading>
              <PanelCollapse
                bsClass="panel"
              >
                <PanelBody
                  bsClass="panel"
                  collapsible={false}
                >
                  <CreateArgumentOptions
                    handleSelect={[Function]}
                  />
                </PanelBody>
              </PanelCollapse>
              <ExplanationPreviewModal
                closeCallbackFn={[Function]}
                container={[Circular]}
                showExplanation={false}
              />
            </Uncontrolled(Panel)>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
        Symbol(enzyme.__setState__): [Function],
      }
    }
    showExplanation={false}
  />
</Uncontrolled(Panel)>
`;
